src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -162,25 +168,40 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
                             BaseAddress = ntModuleInfo.BaseOfDll
                         };
 
-                        int length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length);
+                        int length = 0;
+                        while ((length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length)) == chars.Length)
+                        {
+                            ArrayPool<char>.Shared.Return(chars);
+                            chars = ArrayPool<char>.Shared.Rent(length * 2);
+                        }
+
                         if (length == 0)
                         {
-                            HandleLastWin32Error();
+                            HandleLastWin32Error(module);
                             continue;
                         }
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        while ((length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length)) == chars.Length)
+                        {
+                            ArrayPool<char>.Shared.Return(chars);
+                            chars = ArrayPool<char>.Shared.Rent(length * 2);
```

Comment:
```suggestion
                            char[] toReturn = chars;
                            chars = ArrayPool<char>.Shared.Rent(length * 2);
                            ArrayPool<char>.Shared.Return(chars);
```

Reply:
(@pgovind @tannergooding, growing with the array pool has become pretty common.  We might want to consider adding some kind of `void Resize(ref T[] array, int minimumLength, bool clearArray = false)` API.)